pkg/types/azure/machinepool.go

```diff
@@ -38,4 +44,18 @@ func (a *MachinePool) Set(required *MachinePool) {
 	if required.OSDisk.DiskSizeGB != 0 {
 		a.OSDisk.DiskSizeGB = required.OSDisk.DiskSizeGB
 	}
+
+	a.OSDisk.DiskType = validateDiskType(required.OSDisk.DiskType)
+}
+func validateDiskType(diskType string) string {
+	if diskType == "" {
+		return "Premium_LRS"
+	}
+	_, found := map[string]string{"Standard_LRS": "", "StandardSSD_LRS": "", "Premium_LRS": ""}[ //"UltraSSD_LRS": "",
```

Comment:
This is hard to read. `UltraSSD_LRS` even if commented out should not be within the indexing operator.

Reply:
gofmt formatted it that way when I tried to create a new line and comment it. I think I'll move it to the top of this line.